tiny_dnn/util/product.h

```diff
@@ -516,6 +551,66 @@ void fill(T *dst, size_t size, T value) {
 
 }  // namespace detail
 
+#ifdef CNN_USE_AVX
+// vertically accumulate 'n' AVX registers into single register.
+template <typename aligned>
+#ifdef CNN_USE_DOUBLE
+CNN_MUSTINLINE VECTORIZE_TYPE::register_type accumulate(const double *start,
+                                                        const size_t &nblocks) {
+#else
+CNN_MUSTINLINE VECTORIZE_TYPE::register_type accumulate(const float *start,
+                                                        const size_t &nblocks) {
+#endif
+  const size_t sz = VECTORIZE_TYPE::unroll_size;
```

Comment:
I'd rename this variable to `vsz` or I'd rather use `VECTORIZE_TYPE::unroll_size` directly in `VECTORIZE_TYPE::load` statements.